### PR TITLE
Reload cells when showing the view to fix up any offscreen layout issues

### PR DESCRIPTION
**Pull request by coreyfloyd**
<strike>This didn't fix the issue. Restoring previous behavior plus a little boyscout.

Will update the ticket with the current findings, but the cause is actually tricky to figure out.
</strike>

This fixes the "scroll on save" issue and save button hit area:

- [T125092](https://phabricator.wikimedia.org/T125092)
- [T124976](https://phabricator.wikimedia.org/T124976)

### CODE REVIEW

**Comment by bgerstle** [comment]
@coreyfloyd :-1: still seeing the issue:

> Downloaded saved page: <MWKTitle: 0x1427c90a0> {
    site = "<MWKSite: 0x141089d00> {\n    domain = \"wikipedia.org\";\n    language = en;\n}";
    text = "Neferirkare Kakai";
}
2016-01-28 11:16:08:894 Wikipedia Debug[org.wikipedia.savedarticlesarticleFetcher.accessQueue] -[SavedArticlesFetcher notifyDelegateIfFinished]#L307 INFO:
Finished downloading all saved pages!

but: 

screen shot 2016-01-28 at 11 17 04 am

**Comment by bgerstle** [comment]
Note also that the "Save for later" text isn't showing up for some reason

**Comment by coreyfloyd** [comment]
@bgerstle this is ready for re-review - essentially pulling the fix… need to find another way to address it.

File: Wikipedia/Code/WMFArticleListTableViewController.m [hunk 1]
```diff
@@ -142,12 +143,15 @@ - (void)refreshAnyVisibleCellsWhichAreShowingTitle:(MWKTitle*)title {
...
-    [indexPathsToRefresh enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        self.dataSource.cellConfigureBlock([self.tableView cellForRowAtIndexPath:obj], [self.dataSource itemAtIndexPath:obj], self.tableView, obj);
+    self.dataSource.rowAnimation = UITableViewRowAnimationNone;
+    [indexPathsToRefresh enumerateObjectsUsingBlock:^(id _Nonnull obj, NSUInteger idx, BOOL* _Nonnull stop) {
+        [self.tableView wmf_performUpdates:^{
+            [self.dataSource reloadCellsAtIndexPaths:[self.tableView indexPathsForVisibleRows]];
+        } withoutMovingCellAtIndexPath:[[self.tableView indexPathsForVisibleRows] firstObject]];
     }];
-    
+    self.dataSource.rowAnimation = UITableViewRowAnimationAutomatic;
```
Comment (bgerstle) [inline Wikipedia/Code/WMFArticleListTableViewController.m]: what happens if this is `.None`?

**Comment by coreyfloyd** [comment]
tabling for a bit - might need to only do part of this PR and take the lesser of 2 evils.

**Comment by coreyfloyd** [comment]
@bgerstle one guess i have is that we have some constraints breaking. Maybe the broken constraints are causing the odd scroll behavior

**Comment by bgerstle** [comment]
@coreyfloyd @montehurd think i fixed the issue, as well as the [save button hit area being too small](https://phabricator.wikimedia.org/T125092)

**Comment by bgerstle** [comment]
see video of fix here: https://vid.me/MPuN

**Comment by bgerstle** [comment]
scroll-on-save-fix-demo

**Comment by montehurd** [comment]
:-1: 
- Intrinsic content size has a code smell to me. This shouldn't be needed right?

- Layout issues apparent from looking at visual test screenshots. Way too big touch area if that's what the red outline indicates. Also no longer laying out appropriate vertical space between save button and extract text.

**Comment by montehurd** [comment]
@bgerstle looking good! :+1:

**Comment by montehurd** [comment]
One more with extracts collapsing. BG is fixing.

**Comment by montehurd** [comment]
@bgerstle seems to have fixed it! :+1:

**Pull request merged**